client/my-sites/hosting/controller.js

```diff
@@ -12,15 +12,20 @@ import { get } from 'lodash';
  * Internal Dependencies
  */
 import Hosting from './main';
+import isSiteAutomatedTransfer from 'state/selectors/is-site-automated-transfer';
+import { getSelectedSiteId, getSelectedSite } from 'state/ui/selectors';
 import { isBusinessPlan } from 'lib/plans';
-import { getSelectedSite } from 'state/ui/selectors';
 
-export function redirectIfNotBusiness( context, next ) {
+export function handleHostingPanelRedirect( context, next ) {
 	const { store } = context;
 	const state = store.getState();
+	const isAtomic = isSiteAutomatedTransfer( state, getSelectedSiteId( state ) );
 	const isBusinessSite = isBusinessPlan( get( getSelectedSite( state ), 'plan.product_slug' ) );
 
-	if ( config.isEnabled( 'hosting' ) && isBusinessSite ) {
+	if (
+		config.isEnabled( 'hosting' ) &&
+		( isAtomic || ( config.isEnabled( 'hosting/non-atomic-support' ) && isBusinessSite ) )
+	) {
```

Comment:
I just noticed that it is possible to go direct to the hosting panel if you are a non admin user, eg. just an author - the underlying API checks should stop the user getting any useful data, but I wonder if we should add a check in here for admin user status. If we do, it might be good to pull all this checks into an isHostingEnabled() method?

Reply:
Yeah, perhaps this is the time to surface everything into the api. I'll take a look at doing that today.

Reply:
In the meantime if this check gets too long , we can create a new selector to encapsulate logic.